### ISSUE TITLE
Invalid markup is not marked as such 

### ISSUE THREAD

**Issue by tommhuth** [MananTank/eslint-plugin-validate-jsx-nesting#5]
I do not get warnings for invalid markup such as 

```html
<h2>
    <p>Invalid</p>
</h2>
```
```html
<button> 
    <p>Invalid</p>
</button>
```
 

```html
<a href="/"> 
    <button>Also invalid</button>
</a>
```

Am I missing something here? Is there any limitations as to what kind of invalid markup it catch?

**Comment by MananTank** [comment]
@tommhuth All your examples are considered valid markup. Please refer to https://github.com/MananTank/validate-html-nesting on why that is the caase
